packages/sqs-client/index.ts

```diff
@@ -84,85 +85,242 @@ export type SendSqsMessageBatchParams = {
   messages: SendSqsMessageBatchMessage[];
 };
 
-export const deleteSqsMessage = async (params: DeleteSqsMessageParams) => {
-  const { queueUrl: QueueUrl, receiptHandle: ReceiptHandle } = params;
-  const command = new DeleteMessageCommand({
-    QueueUrl,
-    ReceiptHandle,
-  });
+const S3_MESSAGE_CONTENT_LOCATION_KEY = '__s3MessageLocation';
+
+const MAX_MESSAGE_PAYLOAD_BYTES = 262144;
 
-  return sqs.send(command);
+export const retrieveMessageContent = async (
+  messagePayload: string,
+  messageId?: string,
+): Promise<string> => {
+  const externalContentMessageStart = `${S3_MESSAGE_CONTENT_LOCATION_KEY}=s3://`;
+  console.debug(`Comparing start of message to redirect content
+${externalContentMessageStart}
+${messagePayload.substring(0, externalContentMessageStart.length)}`);
+  if (messagePayload?.startsWith(externalContentMessageStart)) {
```

Comment:
[Nitpick] I find easier to read "early returns", e.g. `if (!messagePayload?.startsWith(externalContentMessageStart)) return messagePayload;`, then the big chunk of logic that handles the other case.